core/src/event_id.rs

```diff
@@ -0,0 +1,66 @@
+use core::fmt::{Display, Formatter};
+
+use miden_crypto::hash::blake::Blake3_256;
+
+use crate::Felt;
+
+/// A type-safe wrapper around a [`Felt`] that represents an event identifier.
+///
+/// Event IDs are used to identify events that can be emitted by the VM or handled by the host.
+/// This newtype provides type safety and ensures that event IDs are not accidentally confused
+/// with other [`Felt`] values.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct EventID(Felt);
```

Comment:
nit: I would rename this into `EventId` - I think this follows Rust conventions better (and we follow a similar approach in other places - e.g., `AccountId`).